syntax = "proto3";

package rollerball;

message Vector3Data {
    float X = 1;
    float Y = 2;
    float Z = 3;
}

message Observation {
    Vector3Data Position = 1;
    Vector3Data TargetPosition = 2;
}

message Action {
    Vector3Data  Force= 1;	
}

message RewardSignal {
    float reward = 1;
    bool done = 2;
}

message PixelData {
    bytes data = 1;
}

// New message type to identify the message content
message UnityMessage {
    oneof message_content {
        Observation observation = 1;
        RewardSignal rewardSignal = 2;
        PixelData pixelData = 3;
    }
}
